### PR TITLE
Add CloseGroup/AbortGroup to mappings

### PR DESCRIPTION
**Pull request by tfpauly**

### CODE REVIEW

**Comment by mwelzl** [comment]
Why this "calling CloseGroup is identical to calling Close" everywhere?  This seems counter-intuitive... am I missing something?   If I create a Connection Group by cloning a Connection a couple of times, no matter what the underlying protocol is, I would expect that calling CloseGroup on one of them closes them all.  This would match the semantics described in the interface draft, section 10.

**Review by mwelzl — CHANGES_REQUESTED** [review]
Sorry, I guess I should have made this a review, not a comment:

Why this "calling CloseGroup is identical to calling Close" everywhere? This seems counter-intuitive... am I missing something? If I create a Connection Group by cloning a Connection a couple of times, no matter what the underlying protocol is, I would expect that calling CloseGroup on one of them closes them all. This would match the semantics described in the interface draft, section 10.

**Comment by tfpauly** [comment]
So, for a multiplexing protocol (SCTP, QUIC), CloseGroup and AbortGroup has a specific meaning to treat the group vs a single connection/stream.

For TCP/UDP I wrote it to be a no-op, but I can say that it affects any other connections that were cloned out? But I wonder if that might be a little weird too.

**Comment by mwelzl** [comment]
I think we need the system to treat it that way for consistency. Clearly, for a mux'ing protocol this is the obvious choice - but the application is not aware of the protocol below... so if it creates a group of TCP connections, with or without congestion control coupling, it will expect:
- that changing a connection property on one of the connections will affect the others (so that also must be implemented - because it WILL happen with a mux'ing protocol),
- that closing the group will close them all.

Imagine an application getting QUIC once, and not getting it another time...  surely we need a consistent behavior towards the app for both of these cases?

File: draft-ietf-taps-impl.md [hunk 1]
```diff
@@ -920,6 +920,12 @@ Close:
 Abort:
 : Calling `Abort` on a TCP Connection indicates that the Connection should be immediately closed by sending a RST to the peer (ABORT.TCP).
 
+CloseGroup:
+: Calling `CloseGroup` on a TCP Connection (CLOSE.TCP) is identical to calling `Close`.
+
+AbortGroup:
+: Calling `AbortGroup` on a TCP Connection (ABORT.TCP) is identical to calling `Abort`.
+
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:928]:
```suggestion
CloseGroup:
: Calling `CloseGroup` on a TCP Connection (CLOSE.TCP) is identical to calling `Close` on all TCP Connections
within a group.

AbortGroup:
: Calling `AbortGroup` on a TCP Connection (ABORT.TCP) is identical to calling `Abort` on all TCP Connections
within a group.

```

File: draft-ietf-taps-impl.md [hunk 2]
```diff
@@ -976,6 +982,12 @@ Close:
 Abort:
 : Calling `Abort` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
 
+CloseGroup:
+: Calling `CloseGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
+
+AbortGroup:
+: Calling `AbortGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
+
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:990]:
```suggestion
CloseGroup:
: Calling `CloseGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all
UDP Connections within a group.

AbortGroup:
: Calling `AbortGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all
UDP Connections within a group.

```

File: draft-ietf-taps-impl.md [hunk 3]
```diff
@@ -1039,6 +1051,12 @@ Close:
 Abort:
 : Calling `Abort` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
 
+CloseGroup:
+: Calling `CloseGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:1055]:
```suggestion
: Calling `CloseGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all UDP Multicast Receive  Connections within a group.
```

File: draft-ietf-taps-impl.md [hunk 4]
```diff
@@ -1039,6 +1051,12 @@ Close:
...
+
+AbortGroup:
+: Calling `AbortGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:1058]:
```suggestion
: Calling `AbortGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`
on all UDP Multicast Receive  Connections within a group.
```

**Comment by tfpauly** [comment]
@mwelzl sounds good, please check the updated text (or suggest text if you have further ideas!)

File: draft-ietf-taps-impl.md [hunk 5]
```diff
@@ -920,6 +920,14 @@ Close:
 Abort:
 : Calling `Abort` on a TCP Connection indicates that the Connection should be immediately closed by sending a RST to the peer (ABORT.TCP).
 
+CloseGroup:
+: Calling `CloseGroup` on a TCP Connection (CLOSE.TCP) is identical to calling `Close` on all TCP Connections
+within a group.
```
Comment (theri) [inline draft-ietf-taps-impl.md:925]: How about:

```suggestion
: Calling `CloseGroup` on a TCP Connection (CLOSE.TCP) is identical to calling `Close` on this Connection and on all Connections in the same ConnectionGroup.
```
This is essentially what it says in Section 10 in the Interface draft, and perhaps the clearest way to phrase this.

**Review by theri — APPROVED** [review]
This is probably fine as is, but I suggested alternative phrasing which might be even more clear.

File: draft-ietf-taps-impl.md (hunk 5 above)
Comment (mwelzl) [reply draft-ietf-taps-impl.md:925]: I like this - I think it should be phrased using that style for all the calls below. "within a group" is just a touch too vague IMO... it needs to be the group this connection is in, even though this is super obvious.

**Review by mwelzl — CHANGES_REQUESTED** [review]
LGTM if the suggestion by @theri is applied everywhere.  Many thanks for doing this BTW!

File: draft-ietf-taps-impl.md [hunk 6]
```diff
@@ -920,6 +920,14 @@ Close:
...
+
+AbortGroup:
+: Calling `AbortGroup` on a TCP Connection (ABORT.TCP) is identical to calling `Abort` on all TCP Connections
+within a group.
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:929]:
```suggestion
AbortGroup:
: Calling `AbortGroup` on a TCP Connection (ABORT.TCP) is identical to calling `Abort` on this Connection and on all Connections in the same ConnectionGroup.
```

File: draft-ietf-taps-impl.md [hunk 7]
```diff
@@ -976,6 +984,14 @@ Close:
...
 : Calling `Abort` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
 
+CloseGroup:
+: Calling `CloseGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all
+UDP Connections within a group.
+
+AbortGroup:
+: Calling `AbortGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all
+UDP Connections within a group.
+
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:994]:
```suggestion
CloseGroup:
: Calling `CloseGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on this Connection and on all Connections in the same ConnectionGroup.

AbortGroup:
: Calling `AbortGroup` on a UDP Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on this Connection and on all Connections in the same ConnectionGroup.
```

File: draft-ietf-taps-impl.md [hunk 8]
```diff
@@ -1039,6 +1055,13 @@ Close:
 Abort:
 : Calling `Abort` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`.
 
+CloseGroup:
+: Calling `CloseGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on all UDP Multicast Receive  Connections within a group.
+
+AbortGroup:
+: Calling `AbortGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`
+on all UDP Multicast Receive  Connections within a group.
+
```
Comment (tfpauly) [inline draft-ietf-taps-impl.md:1064]:
```suggestion
CloseGroup:
: Calling `CloseGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close` on this Connection and on all Connections in the same ConnectionGroup.

AbortGroup:
: Calling `AbortGroup` on a UDP Multicast Receive Connection (ABORT.UDP(-Lite)) is identical to calling `Close`
on this Connection and on all Connections in the same ConnectionGroup.

```

File: draft-ietf-taps-impl.md (hunk 5 above)
Comment (tfpauly) [reply draft-ietf-taps-impl.md]: Applied, thanks!

**Review by mwelzl — APPROVED** [review]
Great, thanks!!

**Review by abrunstrom — APPROVED** [review]
LGTM with the updates, thanks!

**Pull request merged**